source/agora/test/EnrollmentManager.d

```diff
@@ -167,3 +188,133 @@ unittest
         retryFor(node.hasPreimage(enroll.utxo_key,
             EnrollmentManager.PreimageRevealPeriod + 2), 5.seconds));
 }
+
+/// node which refuses to co-operate: doesn't sign / forges the signature / etc
+public extern (C++) class ByzantineNominator : Nominator
+{
+    extern(D) this (KeyPair key_pair, Ledger ledger,
+        TaskManager taskman, NetworkClient[PublicKey] peers,
+        SCPQuorumSet quorum_set)
+    {
+        super(key_pair, ledger, taskman, peers, quorum_set);
+    }
+
+    // refuse to sign
+    extern(C++) override void signEnvelope (ref SCPEnvelope envelope)
+    {
+    }
+}
+
+class BynzantineNode : TestNode
+{
+    this (Config config, Registry* reg)
+    {
+        super(config, reg);
+    }
+
+    override Nominator getNominator (KeyPair key_pair, Ledger ledger,
+        TaskManager taskman, NetworkClient[PublicKey] quorum_peers,
+        SCPQuorumSet quorum_set)
+    {
+        return new ByzantineNominator(key_pair, ledger, taskman,
+            quorum_peers, quorum_set);
+    }
+}
+
+class ByzantineManager : TestAPIManager
+{
+    override void createNewNode (PublicKey address, Config conf)
+    {
+        RemoteAPI!TestAPI api;
+        if (this.nodes.length == 0)  // first node is byzantine
+            api = RemoteAPI!TestAPI.spawn!(BynzantineNode)(conf, &this.reg);
+        else
+            api = RemoteAPI!TestAPI.spawn!(TestNode)(conf, &this.reg);
+
+        this.reg.register(address.toString(), api.tid());
+        this.nodes ~= NodePair(address, api);
+    }
+}
+
+/// Enrollment part of consensus
+unittest
+{
+    import std.algorithm;
+    import std.conv;
+    import core.time;
+
+    TestConf conf = { nodes : 4, threshold : 4 };
+    auto network = makeTestNetwork(TestConf.init);
```

Comment:
You are not passing the `conf` into `makeTestNetwork`, you are passing `TestConf.init`.

Reply:
Is this on purpose?

Reply:
Ah the test still fails, even with that change..

Reply:
Yes, I did a simple test on purpose.

Reply:
Then please remove the `TestConf` instance and also remove the `ByzantineManager` code, unless you intend to use it here. :)